cuda_core/cuda/core/experimental/_program.py

```diff
@@ -489,15 +497,15 @@ def compile(self, target_type, name_expressions=(), logs=None):
 
             return ObjectCode._init(data, target_type, symbol_mapping=symbol_mapping)
 
-        if self._backend == "linker":
+        if self._backend in ("nvJitLink", "driver"):
             return self._linker.link(target_type)
 
```

Comment:
It's a question for the existing code, too:

Is it OK if `None` is returned here?

If yes, I'd make that explicit (`return None`), maybe with a short comment to explain why that's OK. Maybe also mention under `Returns` in the docstring above.

Reply:
No, the 2 if-branches should cover all possibilities, and `compile()` cannot return None. Are you thinking about adding an `assert False` below?

Reply:
Maybe

```
        assert self._backend in ("nvJitLink", "driver")
        return self._linker.link(target_type)
```

?

I'd definitely do *something* to make it obvious that we cannot possibly get a `None` return. (Refactoring accidents happen.)

It could alternative be
```
        raise RuntimeError("This code path is meant to be unreachable.")
```

Reply:
applied in commit 43654c4